app/Http/Controllers/Api/ChannelController.php

```diff
@@ -37,21 +42,43 @@ public function getAllChannels(Request $request, ChannelTransformer $channelTran
     {
         $perPage = $request->query('results') ? : 10;
 
-        $channels = Channel::orderBy('id', 'asc')
+        $channels = Channel::with('episode')->orderBy('channels.id', 'asc')
         ->skip($this->getRecordsToSkip($perPage, $request))
         ->take($perPage)
         ->get();
 
+        $channels = $this->formatChannel($channels);
+
         $resource = new Collection($channels, $channelTransformer);
+
         $data = $this->fractal->createData($resource)->toArray();
 
         if (count($data['data']) > 0) {
             return Response::json($data, 200);
 
         }
 
-        return Response::json(['message' => 'Channels are not available for display'], 404);
+        return Response::json([
+            'message' => 'Channels are not available for display'
+        ], 404);
+
+    }
+
+    public function formatChannel($channels)
+    {
+        foreach ($channels as $key => &$value) {
+            $pending = Channel::pendingEpisodes($value->id)->count();
+            $active  = Channel::activeEpisodes($value->id)->count();
+            $value['user_id'] = UserRepository::findUser($value->user_id)->username;
+            if ($pending != 0 && $active != 0) {
```

Comment:
**Code Quality-Nitpick**
@andela-tolotin The if statement here will only cater for a scenario where a channel has both active and pending episodes, please refactor to cater for other scenario.